### PR TITLE
Headline field showing up in "Footer Contact Info" block

### LINKED ISSUE #4331: Headline field showing up in "Footer Contact Info" block
# Problem
The headline field was made available to the text area block which serves as the base for the Footer Contact Info block. This is not desired.

Screen Shot 2021-10-19 at 10 43 04 AM

# Steps to reproduce
Go to /block/16 and you can see the spurious field.

# Proposed solution

- [x] Update hook to get this particular block and set values to empty
- [x] Form alter to target this block specifically and find a way to remove it (try `#access = FALSE` first). 

See examples of an update hook and form alter where we are targeting this block specifically 
- https://github.com/uiowa/uiowa/pull/4269/files#diff-cce5595154c40be8aea65b9d6fc5d349fc78e0b27205507567745ad5da7a681a
- https://github.com/uiowa/uiowa/pull/4269/files#diff-fead566a578ea17efbda9b82e2a399962f8d578c39998fedf5bb87ceebd7d3f6

### PR DESCRIPTION
**Pull request by dnkemper**
Closes #4331 

# How to test

1. checkout branch
2. `blt ds --site=sandbox.uiowa.edu` (Added a headline that says 'Headline' to the footer in prod for testing)
3.  Make sure the update hook runs (removes existing headline content)
4. Go to https://sandbox.local.drupal.uiowa.edu/block/16?destination=/node/1 and make sure there isn't a headline text field

### CODE REVIEW

**Comment by dnkemper** [comment]
Dang. Copying and pasting function in UI (since merge conflict) didn't copy over correct spaces etc so gotta fix then ready for review again

File: docroot/profiles/custom/sitenow/sitenow.profile [hunk 1]
```diff
@@ -484,6 +484,17 @@ function sitenow_form_alter(&$form, FormStateInterface $form_state, $form_id) {
...
       }
       break;
+
+    // Remove access to headline field in footer contact block.
+    case 'block_content_uiowa_text_area_edit_form':
+      $block = $form_state->getFormObject()->getEntity();
+      $uuid = $block->uuid();
+      // For Footer Contact Information, limit non-admins to minimal.
+      if ($uuid == '0c0c1f36-3804-48b0-b384-6284eed8c67e') {
+        /** @var Drupal\uiowa_core\Access\UiowaCoreAccess $check */
```
Comment (joewhitsitt) [inline docroot/profiles/custom/sitenow/sitenow.profile]: This comment referencing the variable $check doesn't make sense here. It isn't doing a access check/using uiowa_core functionality.

File: docroot/profiles/custom/sitenow/sitenow.install [hunk 2]
```diff
@@ -3553,3 +3553,17 @@ function sitenow_update_9022() {
 function sitenow_update_9023() {
   sitenow_update_9020();
 }
+
+/**
+ * Remove headline content from footer block before setting perms.
+ */
+function sitenow_update_9024() {
+  /** @var \Drupal\Core\Plugin\DefaultPluginManager $filters */
+  $filters = \Drupal::service('plugin.manager.config_filter')->getDefinitions();
```
Comment (joewhitsitt) [inline docroot/profiles/custom/sitenow/sitenow.install]: Is $filters an unused variable? I don't see it called in the code below this

Comment (dnkemper) [reply docroot/profiles/custom/sitenow/sitenow.install]: RIGHT! I was going to call all of the config splits to make sure it was fixed on every site/ or if on core split then call that..  BUT didn't need to. I will remove that

**Review by joewhitsitt — APPROVED** [review]
This worked. Nice. TIL that set() works on compound fields.

This does have the potential of leaving revisions in `block_content_revision__field_uiowa_headline` but I don't think it is a major worry.

Screen Shot 2021-10-21 at 5 24 22 PM

**Comment by dnkemper** [comment]
DANG IT, @joewhitsitt! STOP WORKING!

**Comment by dnkemper** [comment]
Also, thank you so much! Fingers crossed it never becomes an issue 😇

**Pull request merged**